homeassistant/components/mill/sensor.py

```diff
@@ -160,19 +157,19 @@ async def async_setup_entry(
 class MillSensor(CoordinatorEntity, SensorEntity):
     """Representation of a Mill Sensor device."""
 
+    _attr_has_entity_name = True
+
     def __init__(self, coordinator, entity_description, mill_device):
         """Initialize the sensor."""
         super().__init__(coordinator)
 
         self._id = mill_device.device_id
         self.entity_description = entity_description
         self._available = False
-
-        self._attr_name = f"{mill_device.name} {entity_description.name}"
         self._attr_unique_id = f"{mill_device.device_id}_{entity_description.key}"
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, mill_device.device_id)},
-            name=self.name,
+            name=mill_device.name,
```

Comment:
Not really sure how this change affects everything, since before the device would be registered by every sensor under a different name, which doesn't seem right.

Reply:
Mill has moved to a new cloud solution.
So I am working on rewriting the integration. Will make sure the naming is correct in the new version.
https://github.com/home-assistant/core/issues/95424#issuecomment-1627776157

Reply:
Should I wait until that rewrite is done?